File: Amma/Models/Command.swift
import Foundation

enum CommandIntent: String, Codable {
    case castMedia
    case placeCall
    case sendMessage
}

enum CommandStatus: String, Codable {
    case pending
    case executed
    case failed
}

struct Command: Identifiable, Codable {
    let id: UUID
    var familyId: UUID
    var intent: CommandIntent
    var params: [String: String]
    var status: CommandStatus
    var executedAt: Date?
}

File: android/app/src/main/java/com/dhana/amma/models/Command.kt
package com.dhana.amma.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class CommandIntent {
    @SerialName("castMedia") CastMedia,
    @SerialName("stopCast") StopCast,
    @SerialName("placeCall") PlaceCall,
    @SerialName("sendMessage") SendMessage,
}

@Serializable
enum class CommandStatus {
    @SerialName("pending") Pending,
    @SerialName("executed") Executed,
    @SerialName("failed") Failed,
}

@Serializable
data class Command(
    val id: String,
    val familyId: String,
    val intent: CommandIntent,
    val params: Map<String, String> = emptyMap(),
    val status: CommandStatus,
    val executedAt: String? = null,
)
